File: packages/firestore/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/firestore/classes/options/GetCountFromServerOptions.java
package io.capawesome.capacitorjs.plugins.firebase.firestore.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.getcapacitor.JSObject;

import org.json.JSONException;

import io.capawesome.capacitorjs.plugins.firebase.firestore.FirebaseFirestoreHelper;
import io.capawesome.capacitorjs.plugins.firebase.firestore.classes.constraints.QueryCompositeFilterConstraint;

public class GetCountFromServerOptions {

    @NonNull
    private final String reference;

    @Nullable
    private final QueryCompositeFilterConstraint compositeFilter;


    public GetCountFromServerOptions(@NonNull String reference,
                                     @Nullable JSObject compositeFilter) throws JSONException {
        this.reference = reference;
        this.compositeFilter = FirebaseFirestoreHelper.createQueryCompositeFilterConstraintFromJSObject(compositeFilter);
    }

    @NonNull
    public String getReference() {
        return reference;
    }

    @Nullable
    public QueryCompositeFilterConstraint getCompositeFilter() {
        return compositeFilter;
    }
}

File: packages/firestore/ios/Plugin/Classes/Options/GetCountFromServerOptions.swift
import Foundation

@objc public class GetCountFromServerOptions: NSObject {
    private let reference: String
    private let compositeFilter: QueryCompositeFilterConstraint?

    init(reference: String, compositeFilter: JSObject?) {
        self.reference = reference
        self.compositeFilter = FirebaseFirestoreHelper.createQueryCompositeFilterConstraintFromJSObject(compositeFilter)
    }

    func getReference() -> String {
        return reference
    }
    
    func getCompositeFilter() -> QueryCompositeFilterConstraint? {
        return compositeFilter
    }
}
